crates/wasi/src/p3/bindings.rs

```diff
@@ -0,0 +1,306 @@
+        trappable_imports: true,
+        concurrent_exports: true,
+        concurrent_imports: true,
+        async: {
+            only_imports: [
+                "wasi:cli/stdin@0.3.0#get-stdin",
+                "wasi:cli/stdout@0.3.0#set-stdout",
+                "wasi:cli/stderr@0.3.0#set-stderr",
+                "wasi:clocks/monotonic-clock@0.3.0#[async]wait-for",
+                "wasi:clocks/monotonic-clock@0.3.0#[async]wait-until",
+                "wasi:filesystem/types@0.3.0#[method]descriptor.read-via-stream",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.write-via-stream",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.append-via-stream",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.advise",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.sync-data",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.get-flags",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.get-type",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.set-size",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.set-times",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.read-directory",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.sync",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.create-directory-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.stat",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.stat-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.set-times-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.link-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.open-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.readlink-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.remove-directory-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.rename-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.symlink-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.unlink-file-at",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.is-same-object",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.metadata-hash",
+                "wasi:filesystem/types@0.3.0#[async method]descriptor.metadata-hash-at",
+                "wasi:sockets/ip-name-lookup@0.3.0#[async]resolve-addresses",
+                "wasi:sockets/types@0.3.0#[async method]tcp-socket.connect",
+                "wasi:sockets/types@0.3.0#[async method]tcp-socket.send",
+                "wasi:sockets/types@0.3.0#[async method]udp-socket.receive",
+                "wasi:sockets/types@0.3.0#[async method]udp-socket.send",
```

Comment:
For all `[async...]` things, could the original WITs be updated to have `async`?

I realize the is basically the same as my above comment and I also realize that I'm at fault here where I changed the WITs in-repo in wasip3-prototyping, but as we merge upstream this is probably as good a time as any to send PRs to upstream WIT definitions to get `async` in WITs